### PR TITLE
fix: media name and description usage

### PR DESCRIPTION
**Pull request by lacknere**
### 1. Why is this change necessary?
Media name and description are not correctly mapped.

### 2. What does this change do, exactly?
* Fix the hydration and conversion of media name and description.
* Add media name fallbacks if no description is available.

### 6. Checklist

- [x] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: UPGRADE-5.7.md [hunk 1]
```diff
@@ -10,11 +10,18 @@ This changelog references changes done in Shopware 5.7 patch versions.
 
 * Added compatibility with PHP 8.3
 * Added new polyfill `symfony/polyfill-php83` to be able to use PHP 8.3 features
+* Added image name fallbacks if no image description is available to:
```
Comment (mitelg) [inline UPGRADE-5.7.md]:
```suggestion
* Added fallback to image name for `alt` and `title` attributes if image description is not available in following templates:
```

and then please list the tpl files and then the block names

File: engine/Shopware/Bundle/StoreFrontBundle/Gateway/DBAL/Hydrator/MediaHydrator.php [hunk 2]
```diff
@@ -126,8 +126,9 @@ public function hydrateProductImage(array $data)
         $translation = $this->getTranslation($data, '__image');
         $data = array_merge($data, $translation);
 
-        $media->setName($data['__image_description']);
+        $media->setName($data['__image_img']);
```
Comment (mitelg) [inline engine/Shopware/Bundle/StoreFrontBundle/Gateway/DBAL/Hydrator/MediaHydrator.php:129]: would you mind extending the `\Shopware\Tests\Functional\Bundle\StoreFrontBundle\ProductMediaTest`? so we can ensure, this behaviour stays in the future

File: engine/Shopware/Components/Compatibility/LegacyStructConverter.php [hunk 3]
```diff
@@ -641,7 +641,8 @@ public function convertMediaStruct(?Media $media = null)
             'id' => $media->getId(),
             'position' => null,
             'source' => $media->getFile(),
-            'description' => $media->getName(),
+            'name' => $media->getName(),
```
Comment (mitelg) [inline engine/Shopware/Components/Compatibility/LegacyStructConverter.php:644]: please also extend the `\Shopware\Tests\Unit\Components\Compatibility\LegacyStructConverterTest`

File: themes/Frontend/Bare/frontend/index/index.tpl [hunk 4]
```diff
@@ -206,7 +206,10 @@
             {if $sArticle.additionaltext}
                 {$lastSeenProductsConfig.currentArticle.articleName = $lastSeenProductsConfig.currentArticle.articleName|cat:' ':$sArticle.additionaltext}
             {/if}
-            {$lastSeenProductsConfig.currentArticle.imageTitle = $sArticle.image.description}
+            {$lastSeenProductsConfig.currentArticle.imageTitle = $sArticle.image.name}
```
Comment (mitelg) [inline themes/Frontend/Bare/frontend/index/index.tpl:209]: don't we need to escape here as well? :thinking:

**Review by mitelg — CHANGES_REQUESTED** [review]
thanks for your contribution, nice fix! 
would you mind adding a few tests?

File: themes/Frontend/Bare/frontend/index/index.tpl (hunk 4 above)
Comment (lacknere) [reply themes/Frontend/Bare/frontend/index/index.tpl:209]: `$lastSeenProductsConfig`  is json encoded [here](https://github.com/shopware5/shopware/blob/40b94b7027d717cc2b9c9f93a0e1d7a73928232f/themes/Frontend/Bare/frontend/index/index.tpl#L250) so it should be escaped anyways, I think 🤔

**Review by ennasus4sun — APPROVED** [review]

File: engine/Shopware/Bundle/StoreFrontBundle/Gateway/DBAL/Hydrator/MediaHydrator.php (hunk 2 above)
Comment (mitelg) [inline engine/Shopware/Bundle/StoreFrontBundle/Gateway/DBAL/Hydrator/MediaHydrator.php:129]: hey @lacknere 

I had another thought about that. This change is a break. People and code are assuming, that the "name" contains the image description, if this hydration method is used. If we now change this, their code will break. So unfortunately we cannot merge it like that

**Review by mitelg — CHANGES_REQUESTED** [review]

File: engine/Shopware/Bundle/StoreFrontBundle/Gateway/DBAL/Hydrator/MediaHydrator.php (hunk 2 above)
Comment (lacknere) [reply engine/Shopware/Bundle/StoreFrontBundle/Gateway/DBAL/Hydrator/MediaHydrator.php:129]: What about using description primarily and name as a fallback there, if no description is available? Something like `$media->setName($data['__image_description'] ?: $data['__image_img']);`

Comment (mitelg) [reply engine/Shopware/Bundle/StoreFrontBundle/Gateway/DBAL/Hydrator/MediaHydrator.php:129]: sorry for the late answer :see_no_evil: 

I am not sure, if this is also a desired state :thinking: could also lead to unexpected behaviour, where not values where before. 

what about deprecating the old behaviour and trigger the new one with a parameter? :thinking: where the default ensures the old behaviour. 

to not break the public method, we would need to add the parameter commented at first, and get the value with https://www.php.net/manual/en/function.func-get-args.php :thinking: 

maybe you can find a way to build it like that, that you have the possibility to get the hydrated media with "correct" values, but the default way stays at it is.

**Comment by mitelg** [comment]
hey @lacknere 
are you still on this? what are your thoughts?

**Comment by mitelg** [comment]
Hey @lacknere 

I will close this PR due to inactivity. Let me know, if you want to discuss this further :slightly_smiling_face:

**Pull request closed without merging**